### PR TITLE
Fix macro_rules not accepting brackets or parentheses

### PR DESCRIPTION
**Pull request by Veykril**

### CODE REVIEW

**Comment by matklad** [comment]
let's put a [,( test directly over the match arm, as opposed to over the
function

File: crates/parser/src/grammar/items.rs [hunk 1]
```diff
@@ -389,10 +393,12 @@ fn macro_rules(p: &mut Parser, m: Marker) {
     }
 
     match p.current() {
-        T!['{'] => {
+        T!['['] | T!['('] => {
             token_tree(p);
+            p.expect(T![;]);
```
Comment (lnicola) [inline crates/parser/src/grammar/items.rs:398]: Huh, TIL.

Comment (Veykril) [reply crates/parser/src/grammar/items.rs:397]: Only `{}`, `[];` and `();` macro invocations are allowed to produce items, hence you also need the semi here.

**Comment by Veykril** [comment]
bors r=matklad,lnicola

**Pull request merged**